### ISSUE TITLE
Support "fully qualified" operationId that include the handler prefix

### ISSUE THREAD

**Issue by mdwheele** [cdimascio/express-openapi-validator#292]
Hello!

☝️ First, thank you so much for this package. I stumbled across this looking for resources that would promote my application's OpenAPI specification as the "source of truth" for Express as well as docgen and this hits all the marks.

I use [Spotlight](https://stoplight.io) to manage my API specification and am able to specify `operationId`, as usual. However, to use this library, I need to modify source directly to specify handler. This works great, but causes a _bit_ of context switching moving between tooling when I really want to stay focused in Stoplight as much as possible (especially when designing)

It might be cool to be able to specify a "fully qualified" `operationId` (or `x-eov-operation-id`) that is relative to the configured `operationHandlers` path. With this proposal, nothing at all would change about the use of existing options. My thoughts would be this might become a "last attempt" before throwing an `Error`.

From the EOV operations example, imagine the following:

```yaml
paths:
  /ping:
    get:
      description: |
        ping then pong!
      operationId: routes/ping/ping
      # This is no longer required as we can specify the handler in our operationId
      # x-eov-operation-handler: routes/ping 
      responses:
        '200':
          description: OK
          content:
            text/plain:
              schema:
                type: string
                example: pong
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Error'
```

In my own project, I point `operationHandlers` at a `controllers` directory. So my operationId might look more like `auth.login`, `auth.logout` where my handler is situated as:

```js
# controllers/auth.js

module.exports = {
  login: (request, response) => { /* ... */ },
  logout: (request, response) => { /* ... */ },
}
```

What do you think? I would be _glad_ to work on a PR to draft this up if it would help.

**Comment by cdimascio** [comment]
@mdwheele thanks for the ticket and your interest in the project. I like the idea, though I'm a bit hesitant to support it. here's why...

`operationId` is defined by the OpenAPI specification as a unique string used to identify an operation. Tools e.g. code generators commonly use the value of `operationId` as names for a corresponding method.

As such I don't think I should provide `operationId` with new path-like semantics. It may have downstream consequences. One such consequence that results solely on the path like naming means that clients who consume your spec to generate a client or server stub may end with odd path-like method names.

All in all, I understand the pain point you described and surely an optimal workflow would mean modifying code in one place (rather than two). So I'm brainstorming a bit here... but I wonder... One potential option might be to have the Spotlight tools to create the OpenAPI extension i.e.  `x-eov-operation-handler` since it is path aware. I'm not familiar with Spotlight, but since OpenAPI extensions are defined by the specification, its possible Spotlight may already them. Thoughts?

**Comment by cdimascio** [comment]
@mdwheele, please feel free to take a crack at it. I like the ideas.

The following, in particular, is a no-brainer. This should be supported
> Leverage directory-based modules as a convention. In other words, as a convention if no x-eov-operation-handler is specified, try to load ${operationHandlers}/index.js as a "default" object handler.

I like the connexion example. It's clearly working in that ecosystem. There is still a tooling challenge around codegen and the like. Perhaps, it's something reasonably compromised. Ultimately, if one chooses to use complex values for`operationId`, they risk potentially oddities in tooling e.g. codegen, but at the benefit of flexibility elsewhere.

**Comment by mdwheele** [comment]
Hey @cdimascio, I've made some time to work on this. I'm planning on sending a PR early next week depending on how things go. I was able to make progress and everything is working amazingly. I really appreciate the time you've spent building this.

One "side-effect" of standardizing operation handlers (i.e. `some-collection.js` will have a `index`, `show`, `create`, etc. function and so would `some-other-collection.js`) is that Prism (an open-source specification validator) now highlights "Every operation must have a unique 'operationId'":

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/2453394/83289156-9e482400-a1b2-11ea-96da-706757b3f6cf.png">

In a world where `operationId` could be `some-collection.index` or `some-other-collection.create`, we wouldn't run up against this constraint.

**Comment by cdimascio** [comment]
This is great to hear. Thanks for the update. Im willing to accept a PR that handles the path like semantics in operation id. I presume this will enable u to solve the above stated issue. As a start, let's only support forward slashes as path delimiters in operation IDs. Thoughts?

**Comment by mdwheele** [comment]
Hey @cdimascio! I've sent #312 as a conversation starter. I took a slightly different approach that I think will allow users a great deal of flexibility while allowing us to offer a more prescriptive approach to how folks write their schemas.

**Comment by cdimascio** [comment]
this feature is available in`v3.16.0`
thanks @mdwheele for you great work
